blog/2018-07-19-advantages-of-log-based-cdc.adoc

```diff
@@ -0,0 +1,66 @@
+
+All Data Changes Are Captured:: By reading the database's log (e.g. MySQL's binlog or MongoDB's op log) you get the complete list of all data changes in their exact order of application.
+This is vital for many use cases where you are interested in the complete history of record changes.
+In contrast, with a polling-based approach you might miss data changes that happen between two runs of the poll loop.
+
+Low Delays of Events While Avoiding High CPU Load:: With polling, you might be tempted to increase the frequency of polling attempts in order to reduce the chances of missing specific updates.
+While this works to some degree, polling too frequently may cause performance issues (as the queries used for polling cause load on the source database).
+On the other hand, expanding the polling interval will reduce the CPU load but may not only result in missed change events but also in a longer delay for propagating data changes.
+Log-based CDC allows you to react to data changes in near real-time without paying the price of spending CPU time on running polling queries repeatedly.
+
+No Impact on Data Model:: Polling requires some indicator to identify those records that have been changed since the last poll.
+So all the captured tables need to have some column like `LAST_UPDATE_TIMESTAMP` which can be used to find changed rows.
+This can be fine in some cases, but in others such requirement might not be desirable.
+Specifically, you'll need to make sure that the update timestamps are maintained correctly on all tables to be captured by the writing applications or e.g. through triggers.
+
+Can Capture Deletes:: Naturally, polling will not allow you to identify any records that have been deleted since the last poll.
+Often times that's a problem for replication-like use cases where you'd like to have an identical data set on the source database and the replication targets,
+meaning you'd also like to delete records on the sink side if they have been removed in the source database.
+
+Can Capture Old Record State And Further Meta Data:: Depending on the source database's capabilities, log-based CDC can provide the old record state for update and delete events.
+Whereas with polling, you'll only get the current row state.
+Having the old row state handy in a single change event can be interesting for many use cases, e.g. if you'd like to display the complete data change with old and new column values to an application user for auditing purposes.
++
+In addition, log-based approaches often can provide streams of schema changes (e.g. in form of applied DDL statements) and expose additional metadata such as transaction ids or the user applying a certain change.
+These things should generally be doable with query-based approaches, too (depending on the capabilities of the database), I haven't really seen it being done in practice, though.
+
```

Comment:
One more advantage is the ability to catch-up with changes even in case of downtime.

Reply:
Also going to add a mention of downtimes and lost intermediary updates.